### PR TITLE
Micro-optimize GetIdFromArguments to be 48% faster, gaining 0.48s of init time on local (likely more in prod)

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
1. Avoids stringifying key unless its necessary. This was done redundantly twice, but I locked it to just the `isnum` path, as REF will always return a string, and the other path passes `istext`.
2. Use `sortTim` directly instead of `sort_list`. `sort_list` is just sortTim but it copies the list, so it's just wasted cost.

I still would like the bespoke element key option, as that's the only way to drastically cut down costs on things like item descriptions and decals, but this is good for the general use case, and makes it marginally less pressing.

I also want to test if we'd be better off inserting into the list in sorted order rather than sorting it all in the end, but I suspect not.

### CODE REVIEW

File: code/controllers/subsystem/dcs.dm [hunk 1]
```diff
@@ -38,17 +38,18 @@ PROCESSING_SUBSYSTEM_DEF(dcs)
 		var/value
 		if(istext(key))
 			value = arguments[key]
-		if(!(istext(key) || isnum(key)))
+		else if (isnum(key))
+			key = "[key]"
+		else
 			key = REF(key)
-		key = "[key]" // Key is stringified so numbers dont break things
 		if(!isnull(value))
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/dcs.dm:45]: value can only ever be set if the key is text, so this check being done for non string args is redundant 
you may want to include some of <https://github.com/tgstation/tgstation/commit/fa03334c848ea8da61de617cc0a9b0b492b090ba>

Comment (Mothblocks) [reply code/controllers/subsystem/dcs.dm:45]: Looks like with your code you aren't able to use null (or FALSE, but that can be changed with isnull) as a value?

Comment (Mothblocks) [reply code/controllers/subsystem/dcs.dm:45]: well i guess the current code isnt either

Comment (Mothblocks) [reply code/controllers/subsystem/dcs.dm:45]: i eeked out an extra 8% from it, take a look

**Comment by Mothblocks** [comment]
gonna add a unit test to make sure this returns the same ID for several cases as well, to make me a bit less fearful touching this

File: code/controllers/subsystem/dcs.dm [hunk 2]
```diff
@@ -33,22 +33,28 @@ PROCESSING_SUBSYSTEM_DEF(dcs)
...
+				named_arguments[key] = value
+			else
+				fullid += key
+
+			continue
+
+		if (isnum(key))
 			fullid += "[key]"
+		else
+			fullid += REF(key)
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/dcs.dm:54]: I wonder if this means we run REF for paths?

Comment (Mothblocks) [reply code/controllers/subsystem/dcs.dm:54]: yeah, idk what it returns though

Comment (LemonInTheDark) [reply code/controllers/subsystem/dcs.dm:54]: ~~likely the same thing as running REF for strings~~

**Pull request merged**